src/tests/test_magic.py

```diff
@@ -108,6 +108,35 @@ def test_persist(ip):
     assert persisted == [(0, 1, "foo"), (1, 2, "bar")]
 
 
+def test_persist_replace_no_error(ip):
+    runsql(ip, "")
+    ip.run_cell("results = %sql SELECT * FROM test;")
+    ip.run_cell("results_dframe = results.DataFrame()")
+    ip.run_cell("%sql --persist sqlite:// results_dframe")
+
+    out = ip.run_cell("%sql --persist-replace sqlite:// results_dframe")
+    assert out.error_in_exec is None
+
+
+def test_persist_replace_append_replace(ip):
+    runsql(ip, "")
+    ip.run_cell("results = %sql SELECT * FROM test;")
+    ip.run_cell("results_dframe = results.DataFrame()")
+    out = ip.run_cell("%sql --persist-replace --append sqlite:// results_dframe")
+
+    assert isinstance(out.error_in_exec, ValueError)
+
+
+def test_persist_normal_error(ip):
+    runsql(ip, "")
+    ip.run_cell("results = %sql SELECT * FROM test;")
+    ip.run_cell("results_dframe = results.DataFrame()")
+    ip.run_cell("%sql --persist sqlite:// results_dframe")
+    out = ip.run_cell("%sql --persist sqlite:// results_dframe")
+
+    assert isinstance(out.error_in_exec, ValueError)
+
```

Comment:
Let's add another test to check if --persist-replace actually replaced the results

Reply:
Just doing something like this:
```
    ip.run_cell("results = %sql SELECT * FROM test;")
    ip.run_cell("df = results.DataFrame()")
    ip.run_cell("%sql --persist sqlite:// df")
    ip.run_cell("new_results = %sql SELECT * FROM test LIMIT 1;")
    ip.run_cell("df = new_results.DataFrame()")
    ip.run_cell("%sql --persist-replace sqlite:// df")

    out = ip.run_cell("%sql SELECT * FROM df")
    # To assert if the length of out is 1 instead of 2
```